x-pack/plugins/stack_alerts/public/alert_types/geo_containment/readme.md

```diff
@@ -0,0 +1,160 @@
+  key [here](https://docs.google.com/forms/d/e/1FAIpQLSfGUZA6h4eHd2-ImaK5Q_I5Gb7C3UEP5vYDALyGd7r3h08YKg/viewform?hl=en&formkey=dG9kcGIxRFpSS0NhQWM4UjA0V0VkNGc6MQ#gid=0)
+
+### 4. Get trackable point data (MTA bus data) into elasticsearch
+- You'll be using the script: `https://github.com/thomasneirynck/mtatracks` to harvest
+live bus data to populate the system. Clone the repo and follow the instructions in
+the readme to set up. 
+- Using the MTA key you obtained in the previous step, the final command to run
+in a local terminal should look something like the following. This script loads large
+quantities of data the frequency listed below (20000ms = 20s) or higher:
+`node ./load_tracks.js -a <YOUR_API_KEY> -f 20000`
+
+### 5. Open required Kibana tabs
+There are 4 separate tabs you'll need for a combination of loading and viewing the
+data. Since you'll be jumping between them, it might be easiest to just open them
+upfront. Each is preceded by `https://localhost:5601/<your dev env prefix>/app/`:
+- Stack Management > Index Patterns: `management/kibana/indexPatterns`
+- Stack Management > Alerts & Actions: `management/insightsAndAlerting/triggersActions/alerts`
+- Dev tools: `dev_tools#/console`
+- Maps: `maps`
+
+### 6. Create a new alert index in dev tools
+Execute the following two commands:
+- Create index
+```
+PUT /manhattan_mta_alerts
+```
+- Create mappings
+```
+PUT /manhattan_mta_alerts/_mapping
+{
+  "properties": {
+    "entityId": {
+      "type": "keyword"
+    },
+    "entityDateTime": {
+      "type": "date"
+    },
+    "entityDocumentId": {
+      "type": "keyword"
+    },
+    "detectionDateTime": {
+      "type": "date"
+    },    
+    "entityLocation": {
+      "type": "geo_point"
+    },
+    "containingBoundaryId": {
+      "type": "keyword"
+    },
+    "containingBoundaryName": {
+      "type": "keyword"
+    }
+  }
+}
+```
+
+### 7 Create map to monitor alerts
+- Go to the Maps app
+- Create a new map
+- Using GeoJSON Upload, upload the geojson file located in the folder of the previously 
+cloned `mta_tracks` repo: `nyc-neighborhoods.geo.json`. Accept all of the default
+settings.
+- You may want to click your newly added layer and select "Fit to data" so you can see the
+boundaries you've added.
+_ When finished uploading and adding the layer, save the map using a name of your
+choice.
+- Keep the Maps tab open, you'll come back to this
+
+### 8. Create index pattern for generated tracks
+- Go to the index pattern tab to create a new index pattern.
+- Give it the index name `mtatracks*`
+- For `Time field` select `@timestamp`
+- Click `Create index pattern`
+- Leave this tab open, you'll come back to this
+
+
+### 9. Create containment alert
+- Go to the Alerts tab and click `Create Alert` > `Tracking containment`
+- Fill the side bar form top to bottom. This _should_ flow somewhat logically. In the top 
+section, set both `Check every` and `Notify every` to `2 seconds`. For `Notify`, leave
```

Comment:
This is going to starve all other usages of Alerting and Task Manager. If this is a requirement for the containment to work, there's an issue with the implementation of the alert-type.

Reply:
You mean the alerting frequency? It's not a requirement for this alerting type. This is just a "live" demo example with lightning-fast turnarounds so a user can view its capabilities. Setting up a 1-hour interval wouldn't really demo alerts working for a user who wants to quickly test them. We still need to better understand more real-world applications, but intervals will likely be much longer. For cases more similar to this (i.e.- checks every 2 seconds) in the real-world, that's definitely an ongoing conversation with a number of pieces that would need work to really "keep up". All that said, I'll add a note that indicates this is an unusually quick interval. Thanks!

Reply:
Thanks, @aaronjcaldwell. Advising users to not set the alert frequency this high does assuage a lot of my concerns. However, if I understand correctly, the following will cause there to be an action created per vehicle, which also potentially starves other tasks:

> - The default settings for `Select Entity` will mostly be correct. Select `mta_tracks*`
as the index you'd like to track. Use the defaults populated under
`Select entity` > `INDEX`, update `Select entity` > `BY` to `vehicle_ref`.
> - For `Select boundary` > `INDEX`, select `nyc-neighborhoods` and all populated defaults.
> - Under `Actions`, create an `Index` action, then create an `Index connector`. When it
prompts you for a name, you can just call it `index connector` . For index name, use the one we 
created earlier in dev tools: `manhattan_mta_alerts`. Accept the provided defaults and save.

@mikecote do you mind confirming my understanding?

Reply:
@kobelb Unfortunately that is pretty unavoidable at the moment. I think this issue covers this concern (which I also share): https://github.com/elastic/kibana/issues/68828

Reply:
>@mikecote do you mind confirming my understanding?

That's correct. If you're only running one Kibana instance and each execution finds more than 10 documents to index, that will use all the available room Task Manager has and starve other tasks like alerts, actions, scheduled reports, background searches, etc.

Reply:
Until we have https://github.com/elastic/kibana/issues/68828, I don't think that we should use this alert-type in this manner. It's potentially disastrous for other users of Kibana.

Reply:
@kobelb It's worth noting, we're not coming anywhere near the 10 documents limit for this demo alert. This involves < 10  moving fake busses and only alerts on status change, meaning when it only alerts when it first enters a shape. This is far smaller scope than the demo you saw a month or 2 back.

Reply:
After some offline discussion w/ @thomasneirynck, we may remove the map/visual aspect of the demo setup and revise to something simpler to alleviate any confusion on how these alerts can & should be used in the real world. Thanks for the comments!

Reply:
Thanks for your responsiveness and understanding @aaronjcaldwell. I just sent you and a few others a meeting invite for tomorrow for us to ensure we're all on the same page about this. Apologies for dropping the ball and not following up sooner.

Reply:
@kobelb +1 on the concerns wrt. performance.

+1 @aaronjcaldwell original comment why this is in here: the reason this is using the index-connector and the high interval was that internally we were using it because it's much easier to demo when you can talk about it in realtime. Showing these alerts on a map is not the target-use-case for the current implementation. For end-users, the ability to be notified of a vehicle entering an area-of-interest would be most important.

Suggestions to tailor this more to how an actual end-user would end up using this:
- +1 on upping the interval to a higher limit. The containment detection works with wide intervals, in the order of minutes, and will correctly flag "containment". The short intervals just make it easier to demo when explaining what's going on in realtime.
- remove the example of the index-connector. It's only needed if we want to show an alert-instance on the map, which is not something we need to do for GA. Integration in Maps is a future enhancement afaik. maybe use sever-log connector.
- use a more representative boundary-set. iso of generic neighborhoods, a more targeted file that's more indicative of "area-of-interest"; e.g. a  perimeter around UN building or Times-Square
- for the map display to give a sense of "current real world", we can use the existing indices, and do not need to rely on index-connector. e.g. latest-location against the mta-index, and track layer are just as useful to monitor this "current real world" view.

cc @kmartastic

Reply:
Agree with everything @thomasneirynck stated above to change the workflow in the readme.